bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
I had in mind `xlim` behavior, i.e., `factors_bound=['fat', 'cat', 'eat']` shows just those and nothing more. I imagine this kind of bound could be useful in a callback to subset the data without having to reload it?

I'm not sure about `start_bound` and `end_bound`. Remember, it is possible for `start > end` (i.e., if you want "reversed" range displayed) so I think `min` and `max` more clearly express the intent. Then the internals can sort out if `start` or `end` is the min, and do whatever is appropriate. Otherwise we force this on users to know and do the right thing and that seems fragile.